modules/javafx.web/src/main/native/Source/WebCore/platform/java/FileSystemJava.cpp

```diff
@@ -232,21 +232,68 @@ String openTemporaryFile(const String&, PlatformFileHandle& handle)
     return String();
 }
 
-PlatformFileHandle openFile(const String&, FileOpenMode)
+PlatformFileHandle openFile(const String& path, FileOpenMode mode)
 {
-    notImplemented();
-    return invalidPlatformFileHandle;
+    if (mode != FileOpenMode::Read) {
+        return invalidPlatformFileHandle;
+    }
+    JNIEnv* env = WebCore_GetJavaEnv();
+    static jmethodID mid = env->GetStaticMethodID(
+            GetFileSystemClass(env),
+            "fwkOpenFile",
+            "(Ljava/lang/String;Ljava/lang/String;)Ljava/io/RandomAccessFile;");
+    ASSERT(mid);
+
+    JLocalRef<jstring> fileMode = env->NewStringUTF("r");
```

Comment:
Since we support "r" only now, why can't you hard code the mode in FileSystem::fwkOpenFile itself?